aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -756,9 +785,101 @@ Tensor dense_to_sparse_bsr(const Tensor& self, IntArrayRef blocksize) {
 }
 
 Tensor dense_to_sparse_bsc(const Tensor& self, IntArrayRef blocksize) {
-  AT_ERROR(
-      "Conversion from ", self.layout(), " to SparseBsc is currently not supported.");
-  return self;
+  TORCH_CHECK(
+      blocksize[0] > 0 && blocksize[1] > 0,
+      "blocksize needs to be non zero, but got ",
+      blocksize);
+  TORCH_CHECK(
+      self.size(-2) % blocksize[0] == 0,
+      "Tensor size(-2) ",
+      self.size(-2),
+      " needs to be divisible by blocksize[0] ",
+      blocksize[0]);
+  TORCH_CHECK(
+      self.size(-1) % blocksize[1] == 0,
+      "Tensor size(-1) ",
+      self.size(-1),
+      " needs to be divisible by blocksize[1] ",
+      blocksize[1]);
+  auto n_batch_dim = self.dim() - 2;
+  auto is_batched = n_batch_dim > 0;
+  auto values = _batch_tile_tensor(self, blocksize);
+  auto not_zero_mask = _batch_tile_tensor((self != 0), blocksize);
+  not_zero_mask = not_zero_mask.any(-1).any(-1);
+
+  if (is_batched) {
+    // Some extra setup and checks for batched
+    if (n_batch_dim > 1) {
+      // Flatten to single batch dim
+      values = values.flatten(0, n_batch_dim - 1);
+      not_zero_mask = not_zero_mask.flatten(0, n_batch_dim - 1);
+    }
+    TORCH_CHECK(
+        not_zero_mask.size(0) > 0,
+        "to_sparse_bsc: Expected product of batch dimensions to be non-zero.");
```

Comment:
Can this check be moved up before all these `flatten` and non-zero mask manipulations?

Reply:
It could be, but we would have to scan the batch dims to do it (either an `all_of` or `accumulate` and check the result).  The reason it is here is after those manipulations the check is simple. If we move it up the check becomes complicated, and yes we avoid doing the manipulations if we fail the check but I'm not sure optimizing the failure pathway is a good tradeoff.